sdk/iot/Azure.Iot.Hub.Service/src/IoTHubServiceClient.cs

```diff
@@ -81,5 +106,68 @@ public IoTHubServiceClient(Uri endpoint, IoTHubServiceClientOptions options)
+        /// </summary>
+        /// <param name="endpoint">
+        /// The IoT Hub service instance URI to connect to.
+        /// </param>
+        /// <param name="credential">
+        /// The <see cref="TokenCredential"/> implementation which will be used to request for the authentication token.
+        /// </param>
+        public IoTHubServiceClient(Uri endpoint, TokenCredential credential)
+            : this(endpoint, credential, new IoTHubServiceClientOptions())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IoTHubServiceClient"/> class.
+        /// </summary>
+        /// <param name="endpoint">
+        /// The IoT Hub service instance URI to connect to.
+        /// </param>
+        /// <param name="credential">
+        /// The <see cref="TokenCredential"/> implementation which will be used to request for the authentication token.
+        /// </param>
+        /// <param name="options">
+        /// Options that allow configuration of requests sent to the IoT Hub service.
+        /// </param>
+        public IoTHubServiceClient(Uri endpoint, TokenCredential credential, IoTHubServiceClientOptions options)
+        {
+            Argument.AssertNotNull(options, nameof(options));
+
+            _endpoint = endpoint;
+            _clientDiagnostics = new ClientDiagnostics(options);
+
+            options.AddPolicy(new BearerTokenAuthenticationPolicy(credential, GetAuthorizationScopes(_endpoint)), HttpPipelinePosition.PerCall);
+            _httpPipeline = HttpPipelineBuilder.Build(options);
+
+            _registryManagerRestClient = new RegistryManagerRestClient(_clientDiagnostics, _httpPipeline, _endpoint, options.GetVersionString());
+            _twinRestClient = new TwinRestClient(_clientDiagnostics, _httpPipeline, null, options.GetVersionString());
+            _deviceMethodRestClient = new DeviceMethodRestClient(_clientDiagnostics, _httpPipeline, _endpoint, options.GetVersionString());
+
+            Devices = new DevicesClient(_registryManagerRestClient, _twinRestClient, _deviceMethodRestClient);
+            Modules = new ModulesClient(_registryManagerRestClient, _twinRestClient, _deviceMethodRestClient);
+
+            Statistics = new StatisticsClient();
+            Messages = new CloudToDeviceMessagesClient();
+            Files = new FilesClient();
+            Jobs = new JobsClient();
+        }
+
+        /// <summary>
+        /// Gets the scope for authentication/authorization policy.
+        /// </summary>
+        /// <param name="endpoint">The IoT Hub service instance Uri.</param>
+        /// <returns>List of scopes for the specified endpoint.</returns>
+        internal static string[] GetAuthorizationScopes(Uri endpoint)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            Argument.AssertNotNullOrEmpty(endpoint.AbsoluteUri, nameof(endpoint.AbsoluteUri));
+
+            // TODO: GetAuthorizationScopes for IoT Hub
```

Comment:
Pending TODO

Reply:
Do we have tasks for all our TODOs, @vinagesh ?

Reply:
@drwill-ms - I have suggested Abhispsa to setup meetings and do a design review and ask questions to the SDK team wherever appropriate. Our TODOs should come from that and we'll add tasks. Currently we don't have any tasks and some of these probably don't need to be implemented.

Reply:
I think it is best to split the scope of this PR and check in smaller parts that don't need any discussion. @abhipsaMisra - It might be a good idea to break this up into smaller tasks on your card. That way others can also contribute wherever possible. Ex: Azad's script task.

Reply:
We can get a better idea about the sub-tasks after our meeting with Andrei on Monday. The way I see it, if we have to support both sas token and identity based auth, those will definitely be 2 different tasks. 
For the scripts part, Azad and I have synced up offline, he owns a card for this, so there is no overlap. :)

Reply:
You can still add tasks for whatever you are doing. Ex: The task of doing the design review can be a task now. I think it is best to represent the state on the board so that everyone is on the same page and can share work easily.